openedx/core/djangoapps/user_api/accounts/utils.py

```diff
@@ -89,3 +97,77 @@ def _is_valid_social_username(value):
     in the username.
     """
     return '/' not in value
+
+
+def retrieve_last_sitewide_block_completed(username):
+    """
+    Completion utility
+    From a string 'username' or object User retrieve
+    the last course block marked as 'completed' and construct a URL
+
+    :param username: str(username) or obj(User)
+    :return: block_lms_url
+
+    """
+    if not completion_waffle.waffle().is_enabled(completion_waffle.ENABLE_COMPLETION_TRACKING):
+        return
+
+    if not isinstance(username, User):
+        userobj = User.objects.get(username=username)
+    else:
+        userobj = username
+    latest_completions_by_course = BlockCompletion.latest_blocks_completed_all_courses(userobj)
+
+    known_site_configs = [
+        other_site_config.get_value('course_org_filter') for other_site_config in SiteConfiguration.objects.all()
+        if other_site_config.get_value('course_org_filter')
+    ]
+
+    current_site_configuration = get_config_value_from_site_or_settings(
+        name='course_org_filter',
+        site=get_current_site()
+    )
+
+    # courses.edx.org has no 'course_org_filter'
+    # however the courses within DO, but those entries are not found in
+    # known_site_configs, which are White Label sites
+    # This is necessary because the WL sites and courses.edx.org
+    # have the same AWS RDS mySQL instance
+    candidate_course = None
+    candidate_block_key = None
+    latest_date = None
+    # Go through dict, find latest
+    for course, [modified_date, block_key] in latest_completions_by_course.items():
+        if not current_site_configuration:
+            # This is a edx.org
+            if course.org in known_site_configs:
+                continue
+            if not latest_date or modified_date > latest_date:
+                candidate_course = course
+                candidate_block_key = block_key
+                latest_date = modified_date
+
+        else:
+            # This is a White Label site, and we should find candidates from the same site
+            if course.org != current_site_configuration:
```

Comment:
instead of `!=`, shouldn't this be `not in`?

Reply:
@schenedx Yes - good catch. I was testing off of one thing only.